compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -628,6 +628,40 @@ fn check_lhs_nt_follows(sess: &ParseSess, def: &ast::Item, lhs: &mbe::TokenTree)
     // after parsing/expansion. we can report every error in every macro this way.
 }
 
+fn is_empty_token_tree(sess: &ParseSess, seq: &mbe::SequenceRepetition) -> bool {
+    if seq.separator.is_some() {
+        return false;
+    }
+    let mut is_empty = true;
+    let mut iter = seq.tts.iter().peekable();
+    while let Some(tt) = iter.next() {
+        match tt {
+            mbe::TokenTree::MetaVarDecl(_, _, Some(NonterminalKind::Vis)) => {}
+            mbe::TokenTree::Token(t) if matches!(t, Token { kind: DocComment(..), .. }) => {
```

Comment:
You can combine the match and the `matches!` here to one via `mbe::TokenTree::Token(t @ Token { etc })`.